### PR TITLE
 Descriptor should be materialized

### PR DESCRIPTION
**Pull request by q384566678**
When I modify the program in the image-tools, I found that if the Descriptors do not concrete, you can not call it, so I think it should be specific
Signed-off-by: zhouhao zhouhao@cn.fujitsu.com

### CODE REVIEW

File: schema/defs-image.json [hunk 1]
```diff
@@ -15,22 +15,12 @@
...
-          "$ref": "#definitions/mediaType"
-        },
-        "size": {
-          "description": "the size in bytes of the referenced object",
-          "$ref": "defs.json#/definitions/int64"
-        },
-        "digest": {
-          "$ref": "#definitions/digest"
+        "config": {
+          "$ref": "#content-descriptor.json"
```
Comment (wking) [inline schema/defs-image.json]: I'm fine shifting the descriptor properties out into their own property, but manifest-lists currently [don't do that][1].  I think you might be confusing `manifestDescriptor` ([used in the manifest-list][2]) with the [manifest schema][3].

[1]: https://github.com/opencontainers/image-spec/blob/7a6820ed62065744e9142ec93da4b37cc86fed5a/manifest-list.md#manifest-list-property-descriptions
[2]: https://github.com/opencontainers/image-spec/blob/7a6820ed62065744e9142ec93da4b37cc86fed5a/schema/manifest-list-schema.json#L16-L19
[3]: https://github.com/opencontainers/image-spec/blob/7a6820ed62065744e9142ec93da4b37cc86fed5a/schema/image-manifest-schema.json

Comment (q384566678) [inline schema/defs-image.json]: Perhaps I naMaybe I'm a little problem on the naming or understand， but I think in the manifest_list.go, Descriptor is the need for specific, or else when the call will be given the wrong, like“Descriptor has no field or method validate”

Comment (wking) [inline schema/defs-image.json]: It sounds like you're talking about [this Travis error][1] from opencontainers/image-tools#51.  But that's a Go compilation error; I don't see how it relates to the JSON Schema change you're proposing here.

[1]: https://travis-ci.org/opencontainers/image-tools/jobs/170981561#L248

**Comment by philips** [comment]
I don't understand the reason for this change.

**Pull request closed without merging**